### PR TITLE
Widen the tolerations of kuberouter

### LINKED ISSUE #9530: CNI not ready on particular instance group
**1. What `kops` version are you running? The command `kops version`, will display
 this information.**
Version 1.17.0

**2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag.**
Server Version: v1.17.7
Client Version: v1.17.0

**3. What cloud provider are you using?**
Amazon Web Services

**4. What commands did you run?  What is the simplest way to reproduce this issue?**
Created a new cluster running `kops create -f cluster.yaml` , a new secret for SSH key `kops create secret --name test.k8s.local sshpublickey admin -i ~/.ssh/id_rsa.pub` and finally `kops update cluster --yes`

**5. What happened after the commands executed?**
All the resources defined within the YAML file were created, a group of three masters, and two different instance groups. However, the nodes from one instance group were not ready, looking at kubelet logs I could find this: 

Kubelet:
```
Jul 08 20:10:59 ip-172-31-7-154 kubelet[2303]: E0708 20:10:59.830034    2303 kubelet.go:2183] Container runtime network not ready: NetworkReady=false reason:NetworkPluginNotReady message:docker: network plugin is not ready: cni config uninitialized
```

kubeAPIServer logs: 
```
I0708 20:42:59.131972       1 healthz.go:191] [+]ping ok
[+]log ok
[+]etcd ok
[+]poststarthook/start-kube-apiserver-admission-initializer ok
[+]poststarthook/generic-apiserver-start-informers ok
[+]poststarthook/start-apiextensions-informers ok
[+]poststarthook/start-apiextensions-controllers ok
[+]poststarthook/crd-informer-synced ok
[+]poststarthook/bootstrap-controller ok
[-]poststarthook/rbac/bootstrap-roles failed: reason withheld
[+]poststarthook/scheduling/bootstrap-system-priority-classes ok
[+]poststarthook/start-cluster-authentication-info-controller ok
[+]poststarthook/start-kube-aggregator-informers ok
[+]poststarthook/apiservice-registration-controller ok
[+]poststarthook/apiservice-status-available-controller ok
[+]poststarthook/kube-apiserver-autoregistration ok
[+]autoregister-completion ok
[+]poststarthook/apiservice-openapi-controller ok
healthz check failed
```
CNI provider is kube-router. 

**6. What did you expect to happen?**
Bootstrap all the nodes from both instance groups as expected to validate the cluster. 

**7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.**

```
---
apiVersion: kops.k8s.io/v1alpha2
kind: Cluster
metadata:
  name: test.k8s.local
spec:
  kubeAPIServer:
    admissionControl:
      - NamespaceLifecycle
      - LimitRanger
      - ServiceAccount
      - DefaultStorageClass
      - DefaultTolerationSeconds
      - NodeRestriction
      - ResourceQuota
      - AlwaysPullImages
    auditLogMaxAge: 30
    auditLogMaxBackups: 10
    auditLogMaxSize: 100
    auditLogPath: /var/log/kube-apiserver-audit.log
    disableBasicAuth: true
    runtimeConfig:
      autoscaling/v2beta1: "true"
  kubeControllerManager:
    horizontalPodAutoscalerSyncPeriod: 10s
  kubelet:
    anonymousAuth: false
    authorizationMode: Webhook
    authenticationTokenWebhook: true
    featureGates:
      ServiceNodeExclusion: "true"
    evictionHard: memory.available<800Mi
    kubeReserved:
      cpu: 500m
      ephemeral-storage: 1Gi
      memory: 1.6Gi
    kubeReservedCgroup: /podruntime.slice
    readOnlyPort: 0
    systemReserved:
      cpu: 500m
      ephemeral-storage: 1Gi
      memory: 2.4Gi
    systemReservedCgroup: /system.slice
    tlsCipherSuites:
      - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
      - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
      - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305
      - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
      - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305
      - TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
      - TLS_RSA_WITH_AES_256_GCM_SHA384
      - TLS_RSA_WITH_AES_128_GCM_SHA256
  fileAssets:
    - content: |
        [Unit]
        Description=Limited resources slice for Kubernetes services
        Documentation=man:systemd.special(7)
        DefaultDependencies=no
        Before=slices.target
        Requires=-.slice
        After=-.slice
      name: podruntime-slice
      path: /etc/systemd/system/podruntime.slice
  api:
    loadBalancer:
      type: Public
  authorization:
    rbac: {}
  channel: stable
  cloudProvider: aws
  etcdClusters:
  - cpuRequest: 200m
    etcdMembers:
    - instanceGroup: master-us-west-2c-1
      name: "1"
    - instanceGroup: master-us-west-2c-2
      name: "2"
    - instanceGroup: master-us-west-2c-3
      name: "3"
    memoryRequest: 100Mi
    name: main
  - cpuRequest: 100m
    etcdMembers:
    - instanceGroup: master-us-west-2c-1
      name: "1"
    - instanceGroup: master-us-west-2c-2
      name: "2"
    - instanceGroup: master-us-west-2c-3
      name: "3"
    memoryRequest: 100Mi
    name: events
  iam:
    allowContainerRegistry: true
    legacy: false
  kubeDNS:
    provider: CoreDNS
  kubernetesApiAccess:
  - 0.0.0.0/0
  kubernetesVersion: 1.17.7
  networkCIDR: 172.31.0.0/16
  networkID: vpc-XXXXXXXX
  networking:
    kuberouter: {}
  nonMasqueradeCIDR: 100.64.0.0/10
  sshAccess:
  - 0.0.0.0/0
  subnets:
  - cidr: 172.31.0.0/20
    id: subnet-XXXXXXXX
    name: us-west-2c
    type: Public
    zone: us-west-2c
  topology:
    dns:
      type: Public
    masters: public
    nodes: public
---
apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: test.k8s.local
  name: master-us-west-2c-1
spec:
  detailedInstanceMonitoring: true
  machineType: m3.large
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-west-2c-1
  role: Master
  subnets:
  - us-west-2c
---
apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: test.k8s.local
  name: master-us-west-2c-2
spec:
  detailedInstanceMonitoring: true
  machineType: m3.large
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-west-2c-2
  role: Master
  subnets:
  - us-west-2c
---
apiVersion: kops/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: test.k8s.local
  name: master-us-west-2c-3
spec:
  detailedInstanceMonitoring: true
  machineType: m3.large
  maxSize: 1
  minSize: 1
  nodeLabels:
    kops.k8s.io/instancegroup: master-us-west-2c-3
  role: Master
  subnets:
  - us-west-2c
---
apiVersion: kops.k8s.io/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: test.k8s.local
  name: nodes
spec:
  detailedInstanceMonitoring: true
  machineType: c5.2xlarge
  maxSize: 30
  minSize: 4
  role: Node
  nodeLabels:
    kops.k8s.io/instancegroup: nodes
    alpha.service-controller.kubernetes.io/exclude-balancer: "true"
  cloudLabels:
    k8s.io/cluster-autoscaler/enabled: ""
    k8s.io/cluster-autoscaler/k8s1.xxxx.com: ""
  subnets:
  - us-west-2c
---
apiVersion: kops.k8s.io/v1alpha2
kind: InstanceGroup
metadata:
  labels:
    kops.k8s.io/cluster: test.k8s.local
  name: ingress-nodes
spec:
  detailedInstanceMonitoring: true
  additionalSecurityGroups:
  - sg-XXXXXXXX
  externalLoadBalancers:
  - targetGroupArn: arn:aws:elasticloadbalancing:us-west-2:XXXXXXXXXXXX:targetgroup/XXXXX-80/XXXXXX
  - targetGroupArn: arn:aws:elasticloadbalancing:us-west-2:XXXXXXXXXXXX:targetgroup/XXXXX-443/XXXXXX
  - targetGroupArn: arn:aws:elasticloadbalancing:us-west-2:XXXXXXXXXXXX:targetgroup/XXXXX-80/XXXXXX
  - targetGroupArn: arn:aws:elasticloadbalancing:us-west-2:XXXXXXXXXXXX:targetgroup/XXXXX-443/XXXXXX
  machineType: c5.4xlarge
  maxSize: 3
  minSize: 3
  nodeLabels:
    dedicated: ingress
    kops.k8s.io/instancegroup: ingress-nodes
  role: Node
  subnets:
  - us-west-2c
  taints:
  - dedicated=ingress:NoSchedule
```

**8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  Paste the logs into this report, or in a gist and provide the gist link here.**

**9. Anything else do we need to know?**
No

### PR DESCRIPTION
**Pull request by johngmyers**
None of the other CNI DaemonSets pay attention to taints.

Fixes #9530

### CODE REVIEW

File: upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -109,14 +109,7 @@ spec:
...
       tolerations:
-      - key: CriticalAddonsOnly
-        operator: Exists
-      - effect: NoSchedule
-        key: node-role.kubernetes.io/master
-        operator: Exists
-      - effect: NoSchedule
-        key: node.kubernetes.io/not-ready
-        operator: Exists
+      - operator: Exists
```
Comment (hakman) [inline upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template:112]: I think we should go with what Weave has, which seems good enough. As both have similar main contributors could be easily accepted as a patch in upstream.
https://github.com/kubernetes/kops/blob/master/upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template#L251

```suggestion
      - effect: NoSchedule
        operator: Exists
      - effect: NoExecute
        operator: Exists
      - key: CriticalAddonsOnly
        operator: Exists
```

Comment (johngmyers) [reply upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template:112]: I don't see the practical difference between this and what Weave has. And the `CriticalAddonsOnly` one appears to be obsolete, replaced by the `system-node-critical` priorityClassName.

Comment (hakman) [reply upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template:112]: There is no practical difference from my point of view either, just easier to explain a PR for upstream.

**Comment by hakman** [comment]
Probably was not such a good idea to enable testing for kube-router in PRs...
The issues happen probably because of the k8s version changes.

**Comment by johngmyers** [comment]
How can we avoid PRs that break CNIs? Should we have a separate set of CNI presubmits with a stable Kubernetes version?

File: upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template (hunk 1 above)
Comment (johngmyers) [reply upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template:112]: The more succinct version seems clear enough. I'd suggest leading with that for an upstream PR.

Comment (hakman) [reply upup/models/cloudup/resources/addons/networking.kuberouter/k8s-1.12.yaml.template:112]: OK

**Comment by justinsb** [comment]
Let's get this merged.  This is particularly good because we want to replace the term "master" with more inclusive language.  The node selector labels may be trickier, however.

**Pull request merged**